src/extension.ts

```diff
@@ -61,11 +62,17 @@ export class EdgeConfigurationProvider implements vscode.DebugConfigurationProvi
             return null;
         }
 
-        // if there is a version flag, switch to using the new msedge
-        if (config['version'] || config['runtimeExecutable']) {
+        if (msedgeIsOnMachine() || config['version'] || config['runtimeExecutable']) {
             config.type = 'msedge';
         }
 
+        if (config.type == 'edge' && !isEdgeDebuggingSupported()) {
+            const errorMessage = localize('edge.debug.error.versionNotSupported', 'Your version of Microsoft Edge does not support debugging via the Edge DevTools Protocol. You can read more about supported versions here (https://aka.ms/edp-docs).');
+            return vscode.window.showErrorMessage(errorMessage).then(_ => {
```

Comment:
Do we need to add the `.then(...)` call to this? Since we're not passing any options for the user to click, I'd assume this will always return a `Thenable<undefined>` anyway.

Reply:
The problem I run into when I take out the `.then` is that the type checker wants me to handle the case of string also, even though, yes, in our case it will always return a `Thenable<undefined>`
![image](https://user-images.githubusercontent.com/16889261/60356558-25c75a80-9986-11e9-9c90-1d8a14a405ae.png)

Reply:
We can probably add a type hint, e.g. 
```javascript 
<Thenable<undefined>>vscode.window.showErrorMessage(errorMessage);
``` 
since we know it will be undefined, but I'm not too concerned about it, just wanted to understand what was going on 😃 (might be worth adding a comment also).

Thanks!

Reply:
That satisfies the type check! Thank you!! :D